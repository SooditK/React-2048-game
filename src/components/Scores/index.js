import React from 'react';
import PropTypes from 'prop-types';
import styles from './scores.scss';
import i18n from '../../utils/i18n';

export default function Scores({ score, bestScore }) {
  const { index, text, best } = styles;
  return (
    <div className={index}>
      <p className={styles.score}>
        {i18n.score}
        <em className={text}>{score}</em>
      </p>
      <p className={`${styles.score} ${best}`}>
        {i18n.best}
        <em className={text}>{bestScore}</em>
      </p>
    </div>
  );
}

Scores.propTypes = {
  score: PropTypes.number.isRequired,
  bestScore: PropTypes.number.isRequired
};
