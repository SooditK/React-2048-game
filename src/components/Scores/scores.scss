@import '../../assets/styles/vars';

.index {
  color: $gray;
  display: flex;
  font-size: 0.75rem;
  // font-weight: 400;
  justify-content: center;

  .score {
    background: transparent;
    border-radius: 0;
    border-width: 0;
    font-weight: 400;
    margin: 0;
    padding: 1vw 2vw;
    position: relative;
    z-index: 0;

    &::before {
      // background: #fafafa;
      background: rgba(29, 137, 255, 0.9);
      content: '';
      height: 100%;
      left: 0;
      position: absolute;
      top: 0;
      transform: skewX(20deg);
      width: 100%;
      z-index: -1;
    }

    &::after {
      background: rgba(250, 250, 250, 0.3);
      content: '';
      height: 100%;
      left: 0;
      opacity: 0;
      position: absolute;
      top: 0;
      transform: skewX(20deg);
      width: 0;
      z-index: -1;
    }
  }

  .text {
    color: $white;
    display: block;
    font-size: 1.25rem;
    font-style: normal;
    font-weight: 500;
    margin-bottom: -0.3vw;
    padding-top: 0.35vw;
  }

  .best {
    margin-left: 1vw;
  }

  @media all and (max-width: $break-point) {
    .score {
      height: 11vw;
      padding: 3vw 0 2vw;
      width: 20vw;
    }

    .text {
      padding-top: 1.35vw;
    }

    .best {
      margin-left: 2vw;
    }
  }
}
