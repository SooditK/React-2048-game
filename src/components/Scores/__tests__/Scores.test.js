import React from 'react';
import renderer from 'react-test-renderer';
import Scores from '..';

describe('<Scores />', () => {
  it('component render', () => {
    const scores = renderer
      .create(<Scores score={123} bestScore={456} />)
      .toJSON();
    expect(scores).toMatchSnapshot();
  });
});
