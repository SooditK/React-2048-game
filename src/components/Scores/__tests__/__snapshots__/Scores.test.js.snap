// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Scores /> component render 1`] = `
<div
  className="index"
>
  <p
    className="score"
  >
    SCORE
    <em
      className="text"
    >
      123
    </em>
  </p>
  <p
    className="score best"
  >
    BEST
    <em
      className="text"
    >
      456
    </em>
  </p>
</div>
`;
